### ISSUE TITLE
Less verbose installs

### ISSUE THREAD

**Issue by becker33** [LLNL/spack#17464]
Currently, Spack reports curl "errors" when searching for binary packages in the source mirror.

These should instead fail silently, as they are expected behavior.

### Rationale

The current behavior scares users and causes unnecessary bug reports.

We already have `--no-cache` and `--cache-only` options for users who are not okay with packages failing over to source.

**Comment by adamjstewart** [comment]
Do we have a similar issue open for searching for tarballs in the source mirror? For example, when I try to install a newer version of CUDA, I see:
```
==> 131: Installing cuda
==> Finding buildcaches at https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/build_cache
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/build_cache/linux-ubuntu20.04-skylake-gcc-9.3.0-cuda-11.0.2-c6fq5d4vl5hn3cxx7nrrmteeckffhn3i.spec.yaml
curl: (22) The requested URL returned error: 404 Not Found          #          #         #

==> Failed to fetch file from URL: https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/build_cache/linux-ubuntu20.04-skylake-gcc-9.3.0-cuda-11.0.2-c6fq5d4vl5hn3cxx7nrrmteeckffhn3i.spec.yaml
    URL https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/build_cache/linux-ubuntu20.04-skylake-gcc-9.3.0-cuda-11.0.2-c6fq5d4vl5hn3cxx7nrrmteeckffhn3i.spec.yaml was not found!
==> Fetching from https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/build_cache/linux-ubuntu20.04-skylake-gcc-9.3.0-cuda-11.0.2-c6fq5d4vl5hn3cxx7nrrmteeckffhn3i.spec.yaml failed.
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/cuda/cuda-11.0.2
curl: (22) The requested URL returned error: 404 Not Found          #          #         #

==> Failed to fetch file from URL: https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/cuda/cuda-11.0.2
    URL https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/cuda/cuda-11.0.2 was not found!
==> Fetching from https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/cuda/cuda-11.0.2 failed.
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/48/48247ada0e3f106051029ae8f70fbd0c238040f58b0880e55026374a959a69c1
curl: (22) The requested URL returned error: 404 Not Found    #         #          #         #

==> Failed to fetch file from URL: https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/48/48247ada0e3f106051029ae8f70fbd0c238040f58b0880e55026374a959a69c1
    URL https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/48/48247ada0e3f106051029ae8f70fbd0c238040f58b0880e55026374a959a69c1 was not found!
==> Fetching from https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/48/48247ada0e3f106051029ae8f70fbd0c238040f58b0880e55026374a959a69c1 failed.
==> Fetching http://developer.download.nvidia.com/compute/cuda/11.0.2/local_installers/cuda_11.0.2_450.51.05_linux.run
#######################################                                                                                              30.1%
```
This could probably be condensed to a couple of lines.

**Comment by alalazo** [comment]
Can we consider this fixed by #17546 ?
